### PR TITLE
Print GraphiQL Url to Stdout

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Fixes: [#4585](https://github.com/ballerina-platform/ballerina-standard-library/issues/4586)
Fixes: [#4660](https://github.com/ballerina-platform/ballerina-standard-library/issues/4660)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
- [x] Added tests
- [X] Updated the spec
- [X] Checked native-image compatibility
- [X] No `commons` package changes (if there are any, please update the GraphQL version in [GraphQL tools](https://github.com/ballerina-platform/graphql-tools))

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -1652,14 +1652,15 @@ The `graphiql` field is used to provide the GraphiQL client configuration to ena
...
+        path: "/ballerina/graphiql",
+        printUrl: false
     }
 }
 service on new graphql:Listener(9090) {
     // ...
 }
 ```
->**Note:** The field enabled accepts a `boolean` that denotes whether the client is enabled or not. By default, it has been set to `false`. The optional field `path` accepts a valid `string` for the GraphiQL service. If the path is not given in the configuration, `/graphiql` is set as the default path.
+>**Note:** The field `enabled` accepts a `boolean` that denotes whether the client is enabled or not. By default, it has been set to `false`. The optional field `path` accepts a valid `string` for the GraphiQL service. If the path is not given in the configuration, `/graphiql` is set as the default path. The optional field `printUrl` accepts a boolean that denotes whether the GraphiQL url is printed to stdout or not. By default, it has been set to `true`.
```
Comment (ThisaruGuruge) [inline docs/spec/spec.md:1663]: LEt's break down the fields into separate subtitles.

Comment (DimuthuMadushan) [reply docs/spec/spec.md]: Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1524/commits/e5f1c907452289fa565cd7d2cdd0d1bb1cc2c776

File: ballerina/listener.bal [hunk 2]
```diff
@@ -72,6 +77,11 @@ public class Listener {
             if result is error {
                 return error Error("Error occurred while attaching the GraphiQL endpoint", result);
             }
+            if graphiql.printUrl {
+                string sanitizedPath = getSanitizedPath(graphiql.path);
+                string graphiqlUrl = string `http://${self.host}:${self.port}/${sanitizedPath}`;
+                io:println(string `GraphiQL client ready at ${graphiqlUrl}`);
```
Comment (ThisaruGuruge) [inline ballerina/listener.bal:83]: This should be printed inside the `'start` method, not in the attach method. 

Why do we append the `http` manually? What happens when we use SSL for the listener?

Comment (DimuthuMadushan) [reply ballerina/listener.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1524/commits/8cd94860ec2d9d84fb8ec4ef27ecefcad27dd3b8

File: ballerina/listener_utils.bal [hunk 3]
```diff
@@ -414,6 +414,13 @@ isolated function getGraphiqlService(GraphqlServiceConfig? serviceConfig, string
     return graphiqlService;
 }
 
+isolated function getSanitizedPath(string path) returns string {
+    if path.startsWith("/") {
+        return path.substring(1);
+    }
+    return path;
```
[the remark this answers is not in the document]
Comment (DimuthuMadushan) [reply ballerina/listener_utils.bal:418]: Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1524/commits/8cd94860ec2d9d84fb8ec4ef27ecefcad27dd3b8

File: ballerina/listener.bal [hunk 4]
```diff
@@ -41,6 +45,16 @@ public class Listener {
...
         self.wsListener = ();
+        self.graphiql = {};
+        string host = configuration.host;
+        int port = self.httpListener.getPort();
+        if configuration.secureSocket is () {
+            self.httpEndpoint = string `http://${host}:${port}`;
+            self.websocketEndpoint = string `ws://${host}:${port}`;
+        } else {
+            self.httpEndpoint = string `https://${host}:${port}`;
+            self.websocketEndpoint = string `wss://${host}:${port}`;
```
Comment (ThisaruGuruge) [inline ballerina/listener.bal:56]: Do we have a test for the secured WebSocket endpoint?

File: ballerina/listener_utils.bal (hunk 3 above)
[the remark this answers is not in the document]
Comment (ThisaruGuruge) [reply ballerina/listener_utils.bal:418]: Then we can remove this function, right?

File: ballerina/types.bal [hunk 5]
```diff
@@ -58,9 +58,11 @@ public type CorsConfig record {|
 #
 # + enabled - Status of the client
 # + path - Path for the client
+# + printUrl - Whether to enable or disable the GraphiQL URL print
```
Comment (ThisaruGuruge) [inline ballerina/types.bal:61]:
```suggestion
# + printUrl - Enable/disable printing the GraphiQL client URL to the stdout
```

File: docs/spec/spec.md [hunk 6]
```diff
@@ -1652,14 +1655,22 @@ The `graphiql` field is used to provide the GraphiQL client configuration to ena
...
+        path: "/ballerina/graphiql",
+        printUrl: false
     }
 }
 service on new graphql:Listener(9090) {
     // ...
 }
 ```
->**Note:** The field enabled accepts a `boolean` that denotes whether the client is enabled or not. By default, it has been set to `false`. The optional field `path` accepts a valid `string` for the GraphiQL service. If the path is not given in the configuration, `/graphiql` is set as the default path.
+##### 7.1.5.1 Field Enabled
```
Comment (ThisaruGuruge) [inline docs/spec/spec.md:1666]: Check the other places

```suggestion
##### 7.1.5.1 Field `enabled`
```

Comment (ThisaruGuruge) [reply docs/spec/spec.md:1666]: I think the convention we followed in the other places is to use something like this:

```suggestion
##### 7.1.5.1 The `enabled` Field
```

File: ballerina/listener.bal (hunk 4 above)
Comment (DimuthuMadushan) [reply ballerina/listener.bal:56]: Currently we don't have a test

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**